org.rdkit.knime.nodes/src/org/rdkit/knime/nodes/highlighting/RDKitHighlightingNodeModel.java

```diff
@@ -417,6 +418,8 @@ public StreamableOperator createStreamableOperator(PartitionInfo partitionInfo,
 					}
 
 					final MolDraw2DSVG molDrawing = markForCleanup(new MolDraw2DSVG(300, 300), lUniqueWaveId);
+					MolDrawOptions opts = molDrawing.drawOptions();

```

Comment:
Shouldn't this also be wrapped in markForCleanup(..., lUniqueWaveId) ? There is a chance that it gets cleaned up with its parent object MolDraw2DSVG, but if in doubt (and I usually am, because you cannot look into that type of code) I would always call the delete() function at the end. The cleanup code handles it properly, if marked for cleanup.

Reply:
That's a good question, but that object doesn't own any memory of its own... it's just a reference to memory that the MolDraw2DSVG owns, so it should not cause a leak.